### ISSUE TITLE
Delay bar should be shown for every milestone

### ISSUE THREAD

**Issue by RishiRajSahu** [appirio-tech/connect-app#2553]
Whenever there is delay in milestone completion the delay bar should be shown as an alert to notify about the delay. Right now not all milestones support this feature.

**Comment by maxceem** [comment]
@RishiRajSahu some milestones don't have a progress bar which shows the rest time (prgoress), for example `phase-specification`:

![image](https://user-images.githubusercontent.com/146016/46140697-61f94d80-c284-11e8-9588-36bda90714fa.png)

1. Should we show the delay bar only when such milestones without progress bar are delayed:

    ![image](https://user-images.githubusercontent.com/146016/46140798-b4d30500-c284-11e8-8429-5db8de8dca17.png)
2. Or should we add the progress bar to all milestones, so we will show the progress for them even when they are not delayed:

    ![image](https://user-images.githubusercontent.com/146016/46140879-e51aa380-c284-11e8-8eef-a7d43f963d37.png)

**Comment by RishiRajSahu** [comment]
@maxceem 
> Should we show the delay bar only when such milestones without progress bar are delayed

Yes, For milestones without progress bar show it only after being delayed otherwise not.

**Comment by maxceem** [comment]
The full list of components where we have to show a progress bar only when the milestone is delayed:
- [MilestoneTypePhaseSpecification](https://github.com/appirio-tech/connect-app/blob/dev/src/projects/detail/components/timeline/milestones/MilestoneTypePhaseSpecification/MilestoneTypePhaseSpecification.jsx)
- [MilestoneTypeCheckpointReview](https://github.com/appirio-tech/connect-app/blob/dev/src/projects/detail/components/timeline/milestones/MilestoneTypeCheckpointReview/MilestoneTypeCheckpointReview.jsx)
- [MilestoneTypeFinalDesigns](https://github.com/appirio-tech/connect-app/blob/dev/src/projects/detail/components/timeline/milestones/MilestoneTypeFinalDesigns/MilestoneTypeFinalDesigns.jsx)
- [MilestoneTypeDelivery](https://github.com/appirio-tech/connect-app/blob/dev/src/projects/detail/components/timeline/milestones/MilestoneTypeDelivery/MilestoneTypeDelivery.jsx)
- [MilestoneTypeFinalFixes](https://github.com/appirio-tech/connect-app/blob/dev/src/projects/detail/components/timeline/milestones/MilestoneTypeFinalFixes/MilestoneTypeFinalFixes.jsx)

Description of these milestone components and how they can be seen in real life can be found on wiki [here](https://github.com/appirio-tech/connect-app/wiki/Milestones).

**Note**: [MilestoneTypeCheckpointReview](https://github.com/appirio-tech/connect-app/blob/dev/src/projects/detail/components/timeline/milestones/MilestoneTypeCheckpointReview/MilestoneTypeCheckpointReview.jsx) and [MilestoneTypeFinalDesigns](https://github.com/appirio-tech/connect-app/blob/dev/src/projects/detail/components/timeline/milestones/MilestoneTypeFinalDesigns/MilestoneTypeFinalDesigns.jsx) milestones already have progress bar in [old-style usage](https://github.com/appirio-tech/connect-app/wiki/Milestones#old-style-of-milestonetypecheckpointreview-and-milestonetypefinaldesigns) during adding links but don't have progress bar in review mode, thus progress bar has to be added in such case for delayed miletone.

**Comment by gurmeetb** [comment]
@maxceem Kindly assign me this.

**Comment by gurmeetb** [comment]
Can you share sample username/pwd which can be used ?

**Comment by maxceem** [comment]
`phash_manager`/`topcoder123`

**Comment by maxceem** [comment]
As the issue hasn't been completed during 12 hours and no comment was provided, this issue is now open for pick up.

**Comment by gurmeetb** [comment]
ah sorry @maxceem it was quite late yest so couldn’t work on it much. It’s ok i will let know when ready to pickup in few hrs, if it is still open.

**Comment by vigneshTheDev** [comment]
I would like to work on this

**Comment by vigneshTheDev** [comment]
Please review: #2618

**Comment by maxceem** [comment]
> @maxceem Please confirm that MilestoneTypeFinalFixes has this functionality already

Yes, it has to be there as it has a progress bar:

![](https://github.com/appirio-tech/connect-app/wiki/Milestones/images/MilestoneTypeFinalFixes.png)
